### PR TITLE
2571: Add map query params

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Add query params for zoom and poi category filter.

Don't worry, most of the changes are tests :bug: 

### Proposed changes

- Add query param for currently selected poi category (included in share url/pathname)
- Add query param for initial zoom level (not included in share url/pathname)
- Adjust behavior as following:
  - Empty map press: Single and multi poi are deselected and filter modal is closed, filters are retained
  - Selecting map feature (single or multi poi): Previous single or multi pois are deselected, filters are retained
  - Selecting single poi in list: Multi poi and filters are retained
  - Selecting a poi category filter: Single poi and multi pois are deselected
  - Deselecting a poi category filter: Single poi and multi pois are retained
  - Selecting currently open filter: Single poi and multi pois are deselected
  - Deselecting currently open filter: Single poi and multi pois are retained
  - Back to overview button: First single poi is deselected, then multi poi, filters are retained
  - Zoom is never retained (i.e. just initially set via url)
- Extract Pois and PoiFiltersOverlayButtons components (web)
- Fix page title to account for single pois (web)

Does this make sense? Please let me know if you don't understand something or think its not intuitive/good UI/UX.

### Rejected changes

I decided against implementing query params for both currently open filter and filter selection currently shown. Reasoning:
- Currently open filter: The result of this filter is not always the same as it depends on the weekday/time. Therefore, including this in the url could lead to confusing results (i.e. "but it worked on my phone")
- Filter selection shown: I did not see a use case for this and since it would just increase url length and complexity I opted against.

Let me know if you think differently.

### Side effects

None (I think).

**Please test this very thoroughly! I am not sure about quite some behavior here!** :warning:

### Resolved issues

Fixes: #2571.

---

### CODE REVIEW

File: web/src/components/Pois.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import React, { ReactElement, useState } from 'react'
```
Comment (steffenkleinle) [inline web/src/components/Pois.tsx:1]: Mostly just extracted from `PoisPage`

**Review by LeandraH — COMMENTED** [review]
Hmm, when I run `npx uri-scheme open integreat://integreat-app/muenchen/en/locations\?category\=4 --ios`, it opens to the map but no filters are enabled. It's currently not opening at all in Android so can't test there, sorry, I'll try again next week.
Short testing in Firefox worked.

**Comment by steffenkleinle** [comment]
> Hmm, when I run `npx uri-scheme open integreat://integreat-app/muenchen/en/locations\?category\=4 --ios`, it opens to the map but no filters are enabled. It's currently not opening at all in Android so can't test there, sorry, I'll try again next week. Short testing in Firefox worked.

Good catch! Should be fixed :)

**Review by LeandraH — APPROVED** [review]
Very cool! Tested in Firefox and real iOS (still no dice on Android, sorry, someone else will have to test that)

**Comment by steffenkleinle** [comment]
> Very cool! Tested in Firefox and real iOS (still no dice on Android, sorry, someone else will have to test that)

I think you perhaps have to set the app to support the deep link manually (at least for the test cms apk). You can do that in the settings for the app: Open by default > Add link > integreat.app. After that I was able to open https integreat.app links

**Comment by LeandraH** [comment]
> > Very cool! Tested in Firefox and real iOS (still no dice on Android, sorry, someone else will have to test that)
> 
> I think you perhaps have to set the app to support the deep link manually (at least for the test cms apk). You can do that in the settings for the app: Open by default > Add link > integreat.app. After that I was able to open https integreat.app links

You're right, I did, thanks! I even vaguely remember disabling the deep linking for something a while ago 😅  Anyway, worked with that! Very nice!

**Review by f1sh1918 — APPROVED** [review]
Looks fine to me tested on android and chrome.
Not sure why we need the zoom as a query param @steffenkleinle ?
Since it won't be added to the url no one will share it?

**Comment by steffenkleinle** [comment]
> Looks fine to me tested on android and chrome. Not sure why we need the zoom as a query param @steffenkleinle ? Since it won't be added to the url no one will share it?

It was a request by the maltesers to be able to share a map with specific zoom, so like this you can manually add a zoom query param that is then initially set.
See https://chat.tuerantuer.org/digitalfabrik/pl/hsuepxecj38b9mmcpfy4h3us9o

**Pull request merged**